### DISCUSSION TITLE
Early/Snapshot support for post-2024 rule changes

### DISCUSSION THREAD

**Discussion by ebullient** [ebullient/ttrpg-convert-cli#586] [Announcements]
The snapshot build now supports the 2024 ruleset — meaning it won’t crash horribly.

That said, **_it’s not ready_**. I haven’t fully validated the output, and there are warnings and missing content.

### Want to Help?

If you’re comfortable [installing and running the snapshot on your own](https://github.com/ebullient/ttrpg-convert-cli/releases/tag/299-SNAPSHOT), I’d appreciate assistance in these areas:

1. **Validate 2014/older content.** Ensure nothing has changed for users sticking with the old rules.
2. **Identify missing 2024 content.** Some warnings (e.g., missing multiclass data) are already flagged, but a detailed list of other gaps would be helpful.
3. **Create 2024 templates.** Help develop templates, including plain-markdown versions for those not using the fantasy statblock plugin.

### Other Important Changes

- **The `-s` option has been removed.** Sources can now only be specified in the configuration file.
- **Source configuration has changed.** Refer to [the documentation](https://github.com/ebullient/ttrpg-convert-cli/blob/main/docs/configuration.md#specify-content-with-the-sources-key) for details. The [old structure](https://github.com/ebullient/ttrpg-convert-cli/blob/main/docs/configuration.md#migrating-from-full-source-and-convert) will still work.
- **New reprint behavior.** The handling of reprints is now configurable. With the split between 2014 and 2024 content, this is more important. See the updated documentation for details.

### Tips for Using Specific Rulesets

- **To use only 2014 rules:** Exclude xphb and xdmg from your sources. Include "basicrules" and "srd" to constrain SRD material to those versions. See the [Source Map](https://github.com/ebullient/ttrpg-convert-cli/blob/main/docs/sourceMap.md#source-name-mapping-for-5etools).
- **To use only 2024 SRD and free rules:** Add "srd52" and "freerules2024" to your sources. See the [Source Map](https://github.com/ebullient/ttrpg-convert-cli/blob/main/docs/sourceMap.md#source-name-mapping-for-5etools).

**Comment by BlueUser18** [comment]
Still going through the data to confirm 2024 content but after running I did get the multiclass warnings and one other warning for source ESK page 66 no matter what version I run.

![ttrpg-cli_299-SNAPSHOT_Warnings](https://github.com/user-attachments/assets/7ba118d4-d952-4b3d-9e70-68eaa31a1495)

**Reply by ebullient** [reply]
yea.. I have to figure out wtf to do with the multiclass warnings. What I have working for 1.x/2.x is pretty hacky (I knew a lot less about what I needed to do then)

But it is helpful to know that you ran it and it was *mostly* successful. ;)

**Comment by akritikos** [comment]
While running the latest snapshot, all Spells lack the Classes resource. From a quick check, [spellClassIndex](https://github.com/ebullient/ttrpg-convert-cli/blob/7f8f6068e58698b8f18ac16e36a6b66cc860db8f/src/main/java/dev/ebullient/convert/tools/dnd5e/Tools5eIndex.java#L70) appears to never be populated.

Other than that, and the multiclass warnigns already mentioned, I haven't come across anything else looking out of the ordinary!

**Reply by ebullient** [reply]
Fixed in #612

**Reply by akritikos** [reply]
Worked like a charm! 🥳

**Comment by ebullient** [comment]
Released!

https://github.com/ebullient/ttrpg-convert-cli/releases/tag/3.0.0
